Maple2.Database/Storage/Game/GameStorage.Market.cs

```diff
@@ -54,5 +58,59 @@ public partial class Request {
             Context.MesoMarket.Remove(listing);
             return Context.TrySaveChanges();
         }
+
+        public ICollection<PremiumMarketItem> GetPremiumMarketEntries(GenderFilterFlag gender, JobFilterFlag job, string searchString, params int[] tabIds) {
```

Comment:
Were you suggesting caching it in GameSession.State? 
Then just refetching whenever startpage is 1/different tab

Or on the server level?

Reply:
Yea server level, but you can use your own judgement on how to store it. I don't know what operations we need to perform on the data. My instinct is that we cache the exact result from the db, since all post filtering can be derived from that.